crates/solidity/inputs/language/bindings/rules.msgb

```diff
@@ -2060,15 +2060,48 @@ inherit .star_extension
   edge @enum.def -> type
   edge type -> type_enum_type
   edge type_enum_type -> @enum.lexical_scope
-}
 
-@enum [EnumDefinition
+  ; Path to link min and max to the enum's scope (#1158)
+  ; It resolves paths of the form `type(<enum>).min.<something>`
+  node typeof
+  attr (typeof) pop_symbol = "@typeof"
+
+  node built_in_member
+  attr (built_in_member) pop_symbol = "."
+  node min_built_in
+  attr (min_built_in) pop_symbol = "min"
+  node max_built_in
+  attr (max_built_in) pop_symbol = "max"
+  node minmax_type_of
+  attr (minmax_type_of) push_symbol = "@typeof"
+  node minmax_replace
+  attr (minmax_replace) push_symbol = (source-text @name)
```

Comment:
I suggest extracting this path pushing `@typeof -> ENUM_NAME` and re-using it in each member rules below as well. Something like `@enum.value_ref_typeof`. This would be a bit simpler and avoid creating extra nodes for each of the enum elements.

Also not too keen on the `_replace` suffix. Maybe `value_ref_typeof` and `value_ref_type` since both nodes form a reference path to the type of a value of the enum.